contracts/L1/gateway/L1Migrator.sol

```diff
@@ -97,14 +114,25 @@ contract L1Migrator is L1ArbitrumMessenger, IMigrator, EIP712 {
     bytes32 private constant MIGRATE_SENDER_TYPE_HASH =
         keccak256("MigrateSender(address l1Addr,address l2Addr)");
 
+    modifier whenFundsMigrated() {
+        require(ethMigrated && lptMigrated, "FUNDS_NOT_MIGRATED");
```

Comment:
This check will only work if there is a gap between token+lpt migration and delegate migration and someone makes sure that the Tx on L2 succeeded (is redeemed + didn't fail) within that time period